api/v1alpha1/ionoscloudmachine_types.go

```diff
@@ -210,15 +210,36 @@ type Volume struct {
 	AvailabilityZone AvailabilityZone `json:"availabilityZone,omitempty"`
 
 	// Image is the image to use for the VM.
-	//+required
+	//+kubebuilder:validation:XValidation:rule="self.id != '' || has(self.selector)",message="must provide either id or selector"
 	Image *ImageSpec `json:"image"`
 }
 
 // ImageSpec defines the image to use for the VM.
 type ImageSpec struct {
-	// ID is the ID of the image to use for the VM.
+	// ID is the ID of the image to use for the VM. Has precedence over selector.
+	//
 	//+kubebuilder:validation:MinLength=1
```

Comment:
This validation can be removed because the default value is empty string and anything else has a min length of 1